drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -185,6 +194,17 @@ static unsigned int siwx917_on_scan(sl_wifi_event_t event, sl_wifi_scan_result_t
 	if (!sidev->scan_res_cb) {
 		return -EFAULT;
 	}
+
+	if (SL_WIFI_CHECK_IF_EVENT_FAILED(event)) {
```

Comment:
`SL_WIFI_CHECK_IF_EVENT_FAILED()` looks useless. `event & SL_WIFI_EVENT_FAIL_INDICATION` is simpler and clearer for anyone.